clients/src/main/java/org/apache/kafka/clients/consumer/StickyAssignor.java

```diff
@@ -278,18 +303,81 @@
         TreeSet<String> sortedCurrentSubscriptions = new TreeSet<>(new SubscriptionComparator(currentAssignment));
         sortedCurrentSubscriptions.addAll(currentAssignment.keySet());
 
-        balance(currentAssignment, sortedPartitions, unassignedPartitions, sortedCurrentSubscriptions,
+        balance(currentAssignment, prevAssignment, sortedPartitions, unassignedPartitions, sortedCurrentSubscriptions,
                 consumer2AllPotentialPartitions, partition2AllPotentialConsumers, currentPartitionConsumer);
         return currentAssignment;
     }
 
+    private void updatePrevAssignment(Map<TopicPartition, ConsumerGenerationPair> prevAssignment,
+                                      TopicPartition partition, ConsumerGenerationPair consumerGeneration) {
+        if (!prevAssignment.containsKey(partition))
+            prevAssignment.put(partition, consumerGeneration);
+        else {
+            ConsumerGenerationPair existingPair = prevAssignment.get(partition);
+            if (consumerGeneration.generation > existingPair.generation)
+                prevAssignment.put(partition, consumerGeneration);
+            else if (consumerGeneration.generation == existingPair.generation)
+                // same partition is assigned to two consumers during the same rebalance
+                throw new IllegalStateException("Error: Partition '" + partition + "' had been assigned to " +
+                        "multiple consumers following sticky assignment generation " + consumerGeneration.generation + ".");
```

Comment:
Just wondering, if this exception stays (and did not became an error log) because it is impossible to hit and should have been prevented by the previous error log?

Reply:
You're right. I missed replacing this `throw` statement. It's done in the new patch. Thanks!